### PR TITLE
test: Add reworked 2D cornerflow example

### PR DESCRIPTION
**Pull request by adigitoleo**
Also fixes a bug with `misorientation_indices` which caused mixed up M-index outputs and apparent non-reproducible behaviour.

I have removed the old assert statements on this test for now as they don't really test much, also this test is too long for CI anyway and not very tractable as an ensemble test either, so it is mainly qualitative, but at least shows if the M-index is working.

### CODE REVIEW

**Comment by adigitoleo** [comment]
Using the multiprocessed version of `misorientation_indices` instead of looping over `misorientation_index` calls cuts the CPU time in half, but this example is the most memory intensive and needs about 60GB (I've added a note in the docstring).

File: src/pydrex/diagnostics.py [hunk 1]
```diff
@@ -289,7 +289,7 @@ def misorientation_indices(
     if ncpus is None:
         ncpus = len(os.sched_getaffinity(0)) - 1
     with Pool(processes=ncpus) as pool:
-        for i, out in enumerate(pool.imap_unordered(_run, orientation_stack)):
+        for i, out in enumerate(pool.map(_run, orientation_stack)):
```
Comment (Patol75) [inline src/pydrex/diagnostics.py:292]: Just wondering if the large memory usage has anything to do with `map`. See [here](https://docs.python.org/3.11/library/multiprocessing.html#multiprocessing.pool.Pool.map).

**Review by Patol75 — COMMENTED** [review]
Looking great, well done!

File: src/pydrex/diagnostics.py (hunk 1 above)
Comment (adigitoleo) [reply src/pydrex/diagnostics.py:292]: Oh yeah I'll try with `imap` as well, that is still ordered I think.

Comment (adigitoleo) [reply src/pydrex/diagnostics.py]: Switched to `imap`.

**Review by Patol75 — APPROVED** [review]

**Pull request merged**